app/screens/select_server/select_server.js

```diff
@@ -250,7 +255,17 @@ export default class SelectServer extends PureComponent {
 
     onClick = preventDoubleTap(async () => {
         const urlParse = require('url-parse');
-        const preUrl = urlParse(this.state.url, true);
+        let preUrl = urlParse(this.state.url, true);
+        this.attemptWithProtocol = PROTOCOL.HTTPS;
+
+        if (!preUrl.host || preUrl.protocol === 'file:') {
+            preUrl = urlParse('https://' + stripTrailingSlashes(this.state.url), true);
+        }
+
+        if (preUrl.protocol === 'http:') {
```

Comment:
Just to confirm, this means we're always trying https first now even if the user explicitly enters http?